### PR TITLE
Support local flow path

### PR DESCRIPTION
**Pull request by stackptr**
Basing this off of `linter-eslint` to support a local flow path. Adds `findFlowPath` that accepts the file being linted to find `flow` bin in the corresponding modules directory. Need to ensure it still works with a global installation. There are also most likely edge cases that I do not yet understand as the `linter-eslint` code is complicated in this area

### CODE REVIEW

**Comment by jacobwgillespie** [comment]
I am interested in seeing this merged, as this is the only feature I'm missing from other flow packages.  This package, `linter-flow` is the only flow integration for Atom that supports checking flow as-you-type, which is awesome, however unless the global flow version and the project-local version are exactly the same, running one or the other will cause the flow server to restart since the versions don't match.  Practically that means waiting 15s to run flow when switching between the editor and terminal.

Is there anything I can do to help move this PR forward?

**Comment by stackptr** [comment]
Hey @jacobwgillespie this has been working fairly stably for me since I submitted this PR. I'm not really sure if there are further contribution requirements that should be met in order to get this merged. You could try to run this locally as well (I simply symlink `~/.atom/packages/linter-flow` to this cloned branch)

**Comment by stackptr** [comment]
The `flow` binary seems to have some problems when it is spawned multiple times over the course of linting, resulting in CPU locking and other nastiness. I work with an older version of `flow` so I can't speak to whether these problems are fixed currently but I would be interested to know if others would experience this as well. Unfortunately I don't see a fix within scope for this PR

**Comment by stackptr** [comment]
I am using VSCode these days but am interested in doing what I can to finish this enhancement. Can anyone suggest what my next steps should be?

File: lib/helpers.js [hunk 1]
```diff
@@ -2,9 +2,12 @@
 
 'use babel';
 
+import Path from 'path';
+import { findCached } from 'atom-linter';
+
 const helpers = require('atom-linter');
 
-export default function check(
+export function check(
```
Comment (nmn) [inline lib/helpers.js]: I would not change this from a default export to non-default one.

You can still add additional exports while leaving this be the default. I'm trying to minimise breaking changes here.

File: lib/helpers.js [hunk 2]
```diff
@@ -33,3 +36,11 @@ export default function check(
       throw error;
     });
 }
+
+export function findFlowPath(
+  filePath: string,
+): any {
```
Comment (nmn) [inline lib/helpers.js]: return type is string. Don't use `any`

File: lib/index.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ import path from 'path';
 
 import handleData from './message';
 import type { Linter } from './types';
-import check from './helpers';
+import { check, findFlowPath } from './helpers';
```
Comment (nmn) [inline lib/index.js]: Once the check is the default export again, you can use:
`import check, {findFlowPath} from './helpers';`

**Review by nmn — CHANGES_REQUESTED** [review]
I'm sorry I have been inactive for super super long. I want to merge this, but a few small changes are needed. I will also need to test how everything works after so long.

**Comment by nmn** [comment]
@amercier You probably stopped caring about this PR since it's already been a year. If that is the case let me know, and I'll do this myself. If you still want this, feel free to update this PR.

**Comment by amercier** [comment]
@nmn Sorry I didn't author this PR (@stackptr did but, as you said, may not care anymore considering the time). Could you make the changes, please? I (and @hansoksendahl it seems) are still interested :)

**Comment by nmn** [comment]
I will make the changes for now, although the plugin probably needs a big upgrade to use the LSP support built into Flow directly now.

**Comment by stackptr** [comment]
I missed the review on this, I can update this PR soon

**Comment by stackptr** [comment]
Will fix the CI errors soon

**Comment by stackptr** [comment]
@nmn this should be good now

**Pull request closed without merging**